src/routes.py

```diff
@@ -28,11 +37,34 @@ def registration() -> str:
         user.set_password(form.password.data)
         session.add(user)
         session.commit()
+        return redirect(url_for("routes.login"))
     return render_template("registration.html", form=form)
 
 
-@bp.route("/login")
+@bp.route("/login", methods=["POST", "GET"])
 def login() -> str:
     """Handle user's login form."""
+    session_id = ""
     form = LoginForm()
+    if form.validate_on_submit():
+        user = session.query(User).filter_by(username=form.username.data).first()
+        if user and user.check_password(form.password.data):
+            session_id = str(uuid.uuid4())
+            user_session = UserSession(session_id=session_id, user_id=user.id)
+            session.add(user_session)
+            session.commit()
+
+            response = make_response(redirect(url_for("routes.index")))
+            response.set_cookie("session_id", str(session_id))
```

Comment:
Seems like this `str(...)` conversion is redundant here, because we already have this conversion above
```python
session_id = str(uuid.uuid4())
```